gpui: Improve `Background` type

huacnlee opened the pull request:
Release Notes:

- N/A

---

I was changed `Fill` to use `Background` since PR #20812.

But it's not inconvenient to retrieve the background color already set in the current style.

For example: I have use `StyleRefinement` a lot in GPUI Component, I may define a `style` for an element and `impl Styled for MyElement` for allows user to override the element default style.

In this case, I often need to get the `bg` of user sets to do something, e.g.: Adjust that color opacity to border use.

```rs
struct MyElement {
    style: StyleRefinement,
}

impl Styled for MyElement {
    fn style(&mut self) -> &mut StyleRefinement {
        &mut self.style
    }
}

fn render(...) {
    let bg_color = self.style.background;
    // Adjust bg_color ...

    div()
        .bg(default_color)
        .border_color(...)
        .map(|this| this.style().refine(&self.style))
}
```

- [x] Test macOS
- [x] Test Blade macOS
- [x] Test Windows

mikayla-maki reviewed (changes requested):
The presence of this method tells me we don't have the narrative of `Fill` and `Background` settled very well. Why introduce a new word / concept for something we already have? As such, I think we could use a helper, maybe `Background::fill()` to achieve something similar :)

huacnlee committed:
Refactor Background to as enum, and BackgroundColor for GPU internal use.

huacnlee commented:
I changed `Background` to an enum and exposed its members.

I initially added this because the `Background` struct is designed for GPU internals, especially the `tag` property, which isn't suitable for direct use with the GPUI API. Therefore, I intentionally set it to `pub(crate)`.

So now I'm wrapping it in a more easily understood enum type for GPUI users, while the internal structure, renamed `BackgroundColor`, is not publicly accessible.

huacnlee committed:
Update PatternSlash
